crates/rethnet_rpc_server/src/methods/compiler_io.rs

```diff
@@ -0,0 +1,110 @@
+use hashbrown::HashMap;
+
+#[derive(Clone, Debug, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CompilerInput {
+    language: String,
+    /// maps sourceName to content:
+    sources: HashMap<String, CompilerInputSource>,
+    settings: CompilerSettings,
+}
+
+#[derive(Clone, Debug, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CompilerInputSource {
+    content: String,
+}
+
+#[derive(Clone, Debug, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(rename_all = "camelCase")]
+struct CompilerSettings {
+    #[serde(rename = "viaIR")]
+    via_ir: Option<bool>,
+    optimizer: OptimizerSettings,
+    metadata: Option<MetadataSettings>,
+    /// maps a source name to a mapping from contract names to a vector of output
+    /// selections:
+    output_selection: HashMap<String, HashMap<String, Vec<String>>>,
+    evm_version: Option<String>,
+    /// maps a library file name to a mapping from library name to library content:
```

Comment:
```suggestion
    /// Mapping of: file name -> (library name -> library content)
```

I am finding these hard to read. Just an idea, but would the above format make it easier to comprehend?

Reply:
thanks for pushing to improve this. addressed in https://github.com/NomicFoundation/hardhat/pull/3998/commits/5790528b22e19b3bf6ecd0e93d092f35a62c2334